Taarafo.Core.Tests.Unit/Services/Processings/PostImpressions/PostImpressionProcessingServiceTests.Exceptions.RetrieveAll.cs

```diff
@@ -0,0 +1,84 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using System;
+using Moq;
+using Taarafo.Core.Models.Processings.PostImpressions.Exceptions;
+using Xeptions;
+using Xunit;
+
+namespace Taarafo.Core.Tests.Unit.Services.Processings.PostImpressions
+{
+    public partial class PostImpressionProcessingServiceTests
+    {
+        [Theory]
+        [MemberData(nameof(DependencyExceptions))]
+        public void ShouldThrowDependencyExceptionOnRetrieveAllIfDependencyErrorOccursAndLogItAsync(
+            Xeption dependencyException)
+        {
+            //given
+            var somePostImpressions = CreateRandomPostImpressions();
+
+            var expectedPostImpressionProcessingDependencyException =
+                new PostImpressionProcessingDependencyException(
+                    dependencyException.InnerException as Xeption);
+
+            this.postImpressionServiceMock.Setup(service =>
+                service.RetrieveAllPostImpressions()).Throws(dependencyException);
+
+            //when
+            Action retrieveAllPostImpressionAction = () =>
+                this.postImpressionProcessingService.RetrieveAllPostImpressionsAsync();
+
+            //then
+            Assert.Throws<PostImpressionProcessingDependencyException>(
```

Comment:
Change this to:

```
PostImpressionProcessingDependencyException actualPostImpressionProcessingDependencyException = 
    Assert.Throws<PostImpressionProcessingDependencyException>(retrieveAllPostImpressionAction);
```

also do this on the other excptin test

Reply:
Done